bundle/src/main/java/com/adobe/acs/commons/rewriter/impl/TemplatedTransformer.java

```diff
@@ -0,0 +1,146 @@
+        this.aggregatorService = propertyAggregatorService;
+    }
+
+    @Override
+    public void init(ProcessingContext processingContext, ProcessingComponentConfiguration processingComponentConfiguration) throws IOException {
+        SlingHttpServletRequest request = processingContext.getRequest();
+        PageManager pageManager = request.getResourceResolver().adaptTo(PageManager.class);
+
+        // Get the combined properties via service
+        Page page = pageManager.getContainingPage(request.getResource());
+        if (page != null) {
+            properties = aggregatorService.getProperties(page);
+        }
+    }
+
+    public void startElement(String uri, String localName, String quaName, Attributes atts) throws SAXException {
+        if (shouldRun() && localName.equals("a")) {
+            AttributesImpl newAttrs = new AttributesImpl(atts);
+            for (int i = 0; i < newAttrs.getLength(); i++) {
+                String currentAttribute = decode(newAttrs.getValue(i));
+                if (TemplateReplacementUtil.hasPlaceholder(currentAttribute)) {
+
+                    // Get the current placeholder in the string
+                    String placeholder = TemplateReplacementUtil.getPlaceholder(currentAttribute);
+
+                    // Transform it to the key in the property map
+                    String key = TemplateReplacementUtil.getKey(placeholder);
+
+                    // If the placeholder key is in the map then replace it
+                    if (properties.containsKey(key)) {
+                        String replaceValue = (String) properties.get(key);
+                        newAttrs.setValue(i, currentAttribute.replace(placeholder, encode(replaceValue)));
+                    }
+                }
+            }
+            getContentHandler().startElement(uri, localName, quaName, newAttrs);
+        } else {
+            getContentHandler().startElement(uri, localName, quaName, atts);
+        }
+    }
+
+    public void characters(char[] ch, int start, int length) throws SAXException {
+        String currentString = new String(ch);
+        int placeLength = length;
+
+        if (shouldRun()) {
+            String currentPlace = currentString.substring(start, start + length);
+
+            // Get the current placeholders in the string
+            final List<String> placeholders = TemplateReplacementUtil.getPlaceholders(currentPlace);
+
+            for (String placeholder : placeholders) {
+                // Transform it to the key in the property map
+                final String key = TemplateReplacementUtil.getKey(placeholder);
+
+                // If the placeholder key is in the map then replace it
+                if (properties.containsKey(key)) {
+                    final String replaceValue = String.valueOf(properties.get(key));
+                    final String replace = currentPlace.replace(placeholder, replaceValue);
```

Comment:
The code in this function has a lot of very generically named vars that make it hard to fully understand what's happening here.  A couple suggestions:
- `currentPlace` sounds like it would be a numeric position rather than a String - can we name it something more clear?
- `replace` is very generic, and has no cognitive connection to `currentPlace` - whatever we rename `currentPlace` to, can we also then rename `replace` to something that is easily associated?

Reply:
good point - the variable names have been updated